targets/arty/base.py

```diff
@@ -157,6 +157,10 @@ def __init__(self, platform, spiflash="spiflash_1x", **kwargs):
         self.add_memory_region(
             "spiflash", self.mem_map["spiflash"] | self.shadow_base, 16*1024*1024)
 
+        if self.cpu_type == 'vexriscv' and self.cpu_variant == 'linux':
```

Comment:
I don't understand what this is?

Reply:
It definiately requires a discussion.

I proposed some changes to LiteX BIOS booting sequence to make it easier to boot Linux on VexRiscv.
This was my PR: https://github.com/enjoy-digital/litex/pull/216.
It has been merged to master, but then on top of that enjoy-digital added new commits:
* https://github.com/enjoy-digital/litex/commit/af61688d1d86f80210d550fefc24ca247d5754b3
* https://github.com/enjoy-digital/litex/commit/462d12bacc6357b619db9816e31de494358eb96f

As a result VexRiscv emulator requires now it's own memory region described by `EMULATOR_RAM_BASE` or must be located at fixed location 0x50000000.

This modification to `arty/base.py` generates this new memory region.
I believe this should not be arty-specific, but don't know if there is some common code in litex-buildenv where this should go?

Reply:
I've just realized that this is not the proper way of adding new memory region in LiteX. I'll change this to sth like:
```
self.submodules.emulator_ram = wishbone.SRAM(size)                                                                                                                                                                                                                           
self.register_mem('emulator_ram', self.mem_map['emulator_ram'], self.emulator_ram.bus, size)
```